mission/mission_core/missions/ScanTheCode_mission.py

```diff
@@ -0,0 +1,249 @@
+            "stop_model": ["camera_name"]
+        }
+
+        The output gnc_cmd has the following format (does not have to be filled):
+        {
+            "heading": float, # The target heading for the boat to turn towards
+            "vector": tuple, # The target vector for the boat to head towards (will override heading)
+            "poshold": bool, # Whether to set the GNC to poshold mode (heading is not maintained but position is)
+            "waypoint": tuple, # The target waypoint to set the GNC to (sending any other commands will override this)
+            "end_mission": bool # Whether to end the mission
+        }
+
+        '''
+        
+        # Run one yolo inference on the provided frame at camera_data.frame
+        self.storageArray = camera_data.results[0]
+        
+        if(len(self.storageArray) != 0):
+            if(self.debug):
+                print("Interim processing")
+                print(self.storageArray)
+            returnVal = filter(self.storageArray)
+            records = self.analyze(returnVal)
+
+            self.colors, self.pattern = self.results(records)
+            if(self.debug):
+                print("printing results")
+                print(self.colors[0] + self.colors[1] + self.colors[2])
+                print("printing dict outside")
+                print(records)
+            
+            max_vote = records.get(self.pattern)
+            if max_vote is None:
+                max_vote = 0
+            # if(max_vote > 0):
+            #     # Want to update colors for heartbeat
+            #     update_colors_on_server(colors, max_vote, len(storageArray))
+            # else:
+            #     # No Colors to update, can just say black
+            #     update_colors_on_server(["Black", "Black", "Black"], 0, len(storageArray))
+        
+        
+        
+        
+        perc_cmd = {
+            "start": ["center"],
+            "stop": ["port", "starboard"],
+            "record": ["center"],
+            "stop_record": ["port", "starboard"],
+            "load_model": [("center", "ML_Resources/stcA.pt")],
+            "stop_model": ["port", "starboard"]
+        }
+        gnc_cmd = {"poshold": True}
+        return perc_cmd, gnc_cmd
+
+    def end(self):
+        '''
+        Any cleanup code can be placed here.
+        This function is called when the run function returns end_mission = True 
+        or when the mission handler decides to end the mission.
+        '''
+        pass
+    
+    def analyze(stringAnalyzed, debug=False):
+        """Analyzes input string and adds votes to record
+
+        Args:
+            stringAnalyzed (String): raw string of grouped detections
+
+        Returns:
+            dictionary: Record of all votes for each possible pattern
+        """
+        records = {"123":0, "132":0, "213":0, "231":0, "312":0, "321":0 }
+
+        if(debug):    
+            print("Analysis started")
```

Comment:
I updated the template so that instead of prints you should be able to use "self.log()". This is a good catch.

Reply:
Converted all "print" statements to "self.log".